### PR TITLE
Reduce response payload of Calendar API.

### PR DESCRIPTION
**Pull request by JakubCzarlinski**
Also minor changes for inlining code and and reducing await calls.

Previously, the response payload was quite large as all information was being fetched. We only need a couple of fields, so we are now only fetching those. This should reduce the payload size, so reduces the time it takes to fetch the data, so reduces the time it takes to SSR the page.

Also, when we hit a cache miss, we still return the stale results now, but load for the next page refresh. Stale results are fine 99.9% of the time, as the data doesn't change that often.

### CODE REVIEW

File: src/components/calendar/google_api.tsx [hunk 1]
```diff
@@ -9,9 +9,14 @@ const googleCalendar = google.calendar({ version: "v3", auth: apiKey });
 // Refresh events every 5 minutes minus 10 seconds to never force a user to wait.
 var lastUpdated = new Date();
 const rateLimitMinutes = 5;
+const refreshRate = 60000 * rateLimitMinutes - 10000;
+if (refreshRate <= 0) {
+  throw new Error("Rate limit is zero or negative.");
+}
+
```
Comment (JakubCzarlinski) [inline src/components/calendar/google_api.tsx:16]: we do not speak about this skill issue

Comment (AnUnsocialPigeon) [reply src/components/calendar/google_api.tsx:16]: it shall go down in history, never to be repeated...

**Comment by AnUnsocialPigeon** [comment]
I assume this does not break anything? I am honestly unsure how to check lol

**Comment by JakubCzarlinski** [comment]
> I assume this does not break anything? I am honestly unsure how to check lol
> 
> (we check in prod)

yup no breaking changes - only breaking if `refreshRate` is funky but you'd know at point of running locally.

**Review by AnUnsocialPigeon — COMMENTED** [review]
LGTM

**Pull request merged**